src/LibYear/Output/JsonOutput.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Text.Json;
+using LibYear.Core;
+using Spectre.Console;
+
+namespace LibYear.Output;
+
+internal sealed class JsonOutput : IOutput
+{
+	private readonly IAnsiConsole _console;
+
+	public JsonOutput(IAnsiConsole console)
+	{
+		_console = console;
+	}
+
+	public void DisplayAllResults(SolutionResult allResults, bool quietMode)
+	{
+		if (!allResults.Details.Any())
+			return;
+		var model = new ResultOutput(allResults);
+		var serializer = quietMode
+			? IndentedJsonSerializerContext.Default.ResultOutput
+			: FlatJsonSerializerContext.Default.ResultOutput;
+		_console.WriteLine(JsonSerializer.Serialize(model, serializer));
```

Comment:
Can we pull out the serialization into its own (testable) method, and then have the console output what that method returns?

Reply:
Wouldn't it end up being the same single public interface anyway?

Reply:
Yep, `IOutput` would still only have `DisplayAllResults()` but by pulling out what's now `FormatOutput()` we can test that the return value is what we expect without needing to pull it from the test console.